src/Engine/ProtoCore/Lang/BuiltInFunctionEndPoint.cs

```diff
@@ -1151,12 +1154,12 @@ internal static StackValue[] GenerateRangeByAmount(decimal start, long amount, d
                                                          bool hasAmountOp,
                                                          RuntimeCore runtimeCore)
         {
-            double startValue = svStart.ToDouble().RawDoubleValue;
-            double endValue = svEnd.ToDouble().RawDoubleValue;
+            double startValue = svStart.ToDouble().DoubleValue;
+            double endValue = svEnd.ToDouble().DoubleValue;
```

Comment:
I think `ToDouble` calls are not needed, since `DoubleValue` is doing it internally. If so there are quite a number of `ToDouble()` we can remove in this PR.